lib/Core/TerriaError.ts

```diff
@@ -12,32 +14,107 @@ function resolveI18n(i: I18nTranslateString | string) {
   return typeof i === "string" ? i : i18next.t(i.key, i.parameters);
 }
 
+/** Object used to create a TerriaError */
 export interface TerriaErrorOptions {
   /**  A detailed message describing the error.  This message may be HTML and it should be sanitized before display to the user. */
   message: string | I18nTranslateString;
   /** A short title describing the error. */
   title?: string | I18nTranslateString;
+
   /** The object that raised the error. */
   sender?: unknown;
   /** True if the user has seen this error; otherwise, false. */
   raisedToUser?: boolean;
+
+  /** Error which this error was created from. This means TerriaErrors can be represented as a tree of errors - and therefore a stacktrace can be generated */
+  originalError?: TerriaError | Error | (TerriaError | Error)[];
+
+  /** If true, lib\ReactViews\Notification\terriaErrorNotification.tsx will be used to display error message.
+   * If false, a plain old `Notification` will be used
+   */
+  useTerriaErrorNotification?: boolean;
 }
 
+/** Object used to clone an existing TerriaError (see `TerriaError.clone()`).
+ *
+ * If this is a `string` it will be used to set `TerriaError.message`
+ */
+export type TerriaErrorOverrides = Partial<TerriaErrorOptions> | string;
+
 /**
  * Represents an error that occurred in a TerriaJS module, especially an asynchronous one that cannot be raised
  * by throwing an exception because no one would be able to catch it.
  */
 export default class TerriaError {
-  readonly sender: unknown;
   private readonly _message: string | I18nTranslateString;
   private readonly _title: string | I18nTranslateString;
-  raisedToUser: boolean = false;
+  private readonly useTerriaErrorNotification: boolean;
+
+  /** `sender` isn't really used for anything at the moment... */
+  readonly sender: unknown;
+  readonly originalError?: (TerriaError | Error)[];
+  private _raisedToUser: boolean = false;
+
+  /**
+   * Convenience function to generate a TerriaError from some unknown error. It will try to extract a meaningful message from whatever object it is given.
+   *
+   * If error is a `TerriaError`, then it will be `cloned` to create a tree of `TerriaErrors` (see `TerriaError.clone()`).
```

Comment:
```suggestion
   * If error is a `TerriaError`, then `createParentError` it will be used to create a tree of `TerriaErrors` (see {@link `TerriaError#createParentError}`).
```